### PR TITLE
Complex activations

### PR DESCRIPTION
**Pull request by mohame54**
Adds functionality for complex numbers in activation functions, using a new handle_complex_input decorator for **gelu** activation function

### CODE REVIEW

**Comment by mohame54** [comment]
@Ishticode @jshepherd01

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py [hunk 1]
```diff
@@ -169,10 +169,12 @@ def test_jax_leaky_relu(
...
-        available_dtypes=helpers.get_dtypes("float"),
-        large_abs_safety_factor=8,
-        small_abs_safety_factor=8,
-        safety_factor_scale="log",
+        available_dtypes=helpers.get_dtypes("float_and_complex"),
+        large_abs_safety_factor=1,
+        small_abs_safety_factor=1,
+        safety_factor_scale="linear",
+        min_value=-1e4,
+        max_value=1e4,
```
Comment (jshepherd01) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py:177]: What's the reason behind these?

File: ivy/functional/ivy/activations.py [hunk 2]
```diff
@@ -17,19 +17,37 @@
...
+def _gelu_jax_like(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    fn_original=None,
+    approximate: bool = False,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    # We don't have the exact implementation
+    # cuz the erf function doesn't work on complex numbers
```
[the remark this answers is not in the document]
Comment (mohame54) [reply ivy/functional/ivy/activations.py:29]: I tried but It didn't work for me

File: ivy/functional/backends/paddle/activations.py [hunk 3]
```diff
@@ -58,6 +58,9 @@ def leaky_relu(
     return F.leaky_relu(x, negative_slope=alpha)
 
 
+@with_unsupported_device_and_dtypes(
+    {"2.5.1 and below": {"cpu": ("complex128", "complex64")}}, backend_version
+)
```
Comment (jshepherd01) [inline ivy/functional/backends/paddle/activations.py:63]: How come we're marking this as unsupported?

File: ivy/functional/ivy/activations.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mohame54) [reply ivy/functional/ivy/activations.py:29]: I tried to drive another form of erf so we can get it working on complex number but I failed may be later we will find a solution for this problem but to make sure jax has the same problem

File: ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py (hunk 1 above)
Comment (mohame54) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py:177]: the complex numbers  generated with large safety factor were too large and numerically unstable also sometimes It generates nan values  so I decided to lower the min and max value to make the tests consistent across different frameworks

File: ivy/functional/backends/paddle/activations.py (hunk 3 above)
Comment (mohame54) [reply ivy/functional/backends/paddle/activations.py:63]: Which is essential for the approximation of gelu function
```python
def gelu(x): 
    # We don't have the exact implementation
    # cuz the erf function doesn't work on complex numbers
    sqrt_2_over_pi = ivy.sqrt(2 / ivy.pi).astype(x.dtype)
    x_pw = ivy.pow(x, 3)
    cdf = 0.5 * (1.0 + ivy.tanh(sqrt_2_over_pi * (x + 0.044715 * x_pw)))
    return x * cdf
    
```

File: ivy/functional/ivy/activations.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jshepherd01) [reply ivy/functional/ivy/activations.py:29]: Someone on stackexchange has answers, it's still an approximation but its quite a good one, looks very computationally expensive though: https://math.stackexchange.com/a/712568

File: ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py (hunk 1 above)
Comment (jshepherd01) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py:177]: Right but
```
large_abs_safety_factor
    A safety factor of 1 means that all values are included without limitation,
```
the min and max make sense though

File: ivy/functional/backends/paddle/activations.py (hunk 3 above)
Comment (jshepherd01) [reply ivy/functional/backends/paddle/activations.py:63]: Ah, fair, so those limits would then be removed if/when `tanh` gets complex number support?

File: ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py (hunk 1 above)
Comment (mohame54) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py:177]: I know that's odd but it worked for me

Comment (jshepherd01) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_nn/test_non_linear_activations.py:177]: Fair enough

**Pull request merged**